components/LocationInput.tsx

```diff
@@ -8,126 +8,164 @@ import React, {
+  originalLat: number;
+  shouldZoomIntoOriginal: boolean;
   setIsSubmissionEnabled: (isSubmissionEnabled: boolean) => void;
-  setLngExternal: (lng: number) => void;
-  setLatExternal: (lat: number) => void;
+  setMapCoordinates: ({ longitude, latitude }: LatitudeLongitude) => void;
   zoomLevelThreshold: number;
+  initialZoom: number;
+  canEdit: boolean;
 }
 
-const originalLng = -2.3175601;
-const originalLat = 54.70534432;
-
-const LocationInput: FC<PropsWithChildren<LocationInputProps>> = ({
+const LocationInput: FC<LocationInputProps> = ({
+  originalLat,
+  originalLng,
+  shouldZoomIntoOriginal,
+  setMapCoordinates,
   setIsSubmissionEnabled,
-  setLatExternal,
-  setLngExternal,
   zoomLevelThreshold,
+  initialZoom,
+  canEdit,
 }) => {
   mapboxgl.accessToken = process.env.NEXT_PUBLIC_MAPBOX_PUBLIC!;
   const mapContainer = useRef<HTMLDivElement | null>(null);
   const map = useRef<mapboxgl.Map>();
   const geocoderContainer = useRef<HTMLDivElement | null>(null);
-  const [isMapReady, setIsMapReady] = useState(false);
-  const [lng, setLng] = useState<number>(originalLng);
-  const [lat, setLat] = useState<number>(originalLat);
-  const [zoom, setZoom] = useState<number>(5);
+  const [zoom, setZoom] = useState<number>(initialZoom);
 
   useEffect(() => {
     if (map.current) return; // initialize map only once
     if (mapContainer.current) {
       map.current = new mapboxgl.Map({
         container: mapContainer.current,
         style: 'mapbox://styles/alester3/clf1lj7jg000b01n4ya2880gi',
-        center: [lng, lat],
+        center: [originalLng, originalLat],
         zoom,
         keyboard: false,
         attributionControl: false,
+        interactive: canEdit,
       });
 
       const nav = new mapboxgl.NavigationControl({ showCompass: false });
       map.current.addControl(nav, 'bottom-right');
-      map.current.addControl(
-        new mapboxgl.GeolocateControl({
-          positionOptions: {
-            enableHighAccuracy: true,
-          },
-        })
-      );
+
+      // Adds a current location button
+      if (canEdit) {
+        map.current.addControl(
+          new mapboxgl.GeolocateControl({
+            positionOptions: {
+              enableHighAccuracy: true,
+            },
+          })
+        );
+      }
+
+      let popup: mapboxgl.Popup | null = null;
 
       const geocoder = new MapboxGeocoder({
         accessToken: mapboxgl.accessToken,
         mapboxgl: mapboxgl,
-        placeholder: 'Where is your solar panel located?',
+        placeholder: 'Search a location',
         marker: false,
         reverseGeocode: true,
       });
-      if (geocoderContainer.current) {
-        geocoderContainer.current.appendChild(geocoder.onAdd(map.current));
+
+      if (!canEdit) {
+        // Prevent user from changing the geocoder input when the map isn't editable
+        geocoder.setBbox([0, 0, 0, 0]);
       }
 
+      geocoderContainer.current?.appendChild(geocoder.onAdd(map.current));
```

Comment:
According to the designs, we should be adding the geocoder on top of the map. However, I think this is fine for now—the position of the geocoder above the map might work better and look better in the long run. We do still need to style the geocoder according to the designs though. Could you try to do this using the styles in `globals.css`?